bundles/org.openhab.binding.smsmodem/src/main/java/org/openhab/binding/smsmodem/internal/handler/SMSModemBridgeHandler.java

```diff
@@ -0,0 +1,487 @@
+import org.slf4j.LoggerFactory;
+import org.smslib.CommunicationException;
+import org.smslib.Modem;
+import org.smslib.Modem.Status;
+import org.smslib.callback.IDeviceInformationListener;
+import org.smslib.callback.IInboundOutboundMessageListener;
+import org.smslib.callback.IModemStatusListener;
+import org.smslib.message.AbstractMessage.Encoding;
+import org.smslib.message.DeliveryReportMessage;
+import org.smslib.message.InboundMessage;
+import org.smslib.message.MsIsdn;
+import org.smslib.message.OutboundMessage;
+import org.smslib.message.Payload;
+import org.smslib.message.Payload.Type;
+
+/**
+ * The {@link SMSModemBridgeHandler} is responsible for handling
+ * communication with the modem.
+ *
+ * @author Gwendal ROULLEAU - Initial contribution
+ */
+@NonNullByDefault
+public class SMSModemBridgeHandler extends BaseBridgeHandler
+        implements IModemStatusListener, IInboundOutboundMessageListener, IDeviceInformationListener {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(
+            SMSModemBindingConstants.SMSMODEMBRIDGE_THING_TYPE,
+            SMSModemBindingConstants.SMSMODEMREMOTEBRIDGE_THING_TYPE);
+
+    private final Logger logger = LoggerFactory.getLogger(SMSModemBridgeHandler.class);
+
+    private SerialPortManager serialPortManager;
+
+    /**
+     * The smslib object responsible for the serial communication with the modem
+     */
+    private @Nullable Modem modem;
+
+    /**
+     * A scheduled watchdog check
+     */
+    private @Nullable ScheduledFuture<?> checkScheduled;
+
+    // we keep a list of msisdn sender for autodiscovery
+    private Set<String> senderMsisdn = new HashSet<String>();
+    private @Nullable SMSConversationDiscoveryService discoveryService;
+
+    private boolean shouldRun = false;
+
+    public SMSModemBridgeHandler(Bridge bridge, SerialPortManager serialPortManager) {
+        super(bridge);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void dispose() {
+        shouldRun = false;
+        ScheduledFuture<?> checkScheduledFinal = checkScheduled;
+        if (checkScheduledFinal != null) {
+            checkScheduledFinal.cancel(true);
+        }
+        Modem finalModem = modem;
+        if (finalModem != null) {
+            scheduler.execute(finalModem::stop);
+            finalModem.registerStatusListener(null);
+            finalModem.registerMessageListener(null);
+            finalModem.registerInformationListener(null);
+        }
+        modem = null;
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        scheduler.execute(() -> {
+            Modem finalModem = modem;
+            if (finalModem != null) {
+                finalModem.stop();
+            }
+            checkAndStartModemIfNeeded();
+        });
+    }
+
+    @Override
+    public void initialize() {
+        shouldRun = true;
```

Comment:
This is necessary to make the thing status automatism work.
```suggestion
        updateStatus(ThingStatus.UNKNOWN);
        shouldRun = true;
```